Clarify image of text rule (#1969)
auto-wcag/auto-wcag: _rules/image-no-text-0va7u6.md

## Applicability

This rule applies to any [image resources][rendered image resources] rendered in a [web page][].

## Expectation

Each test target has no [visible][] [text][human language], except if at least one of the following is true:

- <dfn id="0va7u6:decorative">decorative</dfn>: The image with text is [purely decorative][]; or
- <dfn id="0va7u6:incidental">incidental</dfn>: The text is not a [significant][insignificant] part of the image; or

> - <dfn id="0va7u6:essential">essential</dfn>: Ensuring consistent presentation of the text is [essential][].

Reviewer: With the definitions here, I think that example descriptions should refer to them where relevant.
For example, Passed example 4 (Time Square):
> `This image resource referenced by the `object` element contains text, but it is not the most significant content.`
should be:
> `This image resource referenced by the `object` element contains text, but it is [incidental](#0va7u6:incidental).`

> - <dfn id="0va7u6:essential">essential</dfn>: Ensuring consistent presentation of the text is [essential][].

Reviewer 2: Do we need to say it is "consistent"? Consistent with what? 

```suggestion
- <dfn id="0va7u6:essential">essential</dfn>: Ensuring presentation of the text is [essential][].
```

## Assumptions

- There is no mechanism to change the rendered text in the image resource. Otherwise, the rule might fail while [SC 1.4.5 Images of Text][sc1.4.5] and [SC 1.4.9 Images of Text (No Exception)][sc1.4.9] might be satisfied.
- The specific presentation of the text rendered in the image resource can be achieved through formatted text. Otherwise, the rule might fail while [SC 1.4.5 Images of Text][sc1.4.5] and [SC 1.4.9 Images of Text (No Exception)][sc1.4.9] might be satisfied.

> - When used in HTML, the SVG `<text>` element is not considered to be an image of text. This is because like any other element in HTML, SVG `<text>` can be adjusted through custom style sheets. This is not true for SVG text that is in a separate file, and displayed through for example the `img` element.

Reviewer 2: ```suggestion
- When used in HTML, the SVG `<text>` element is not considered to be an image of text. This is because like any other element in HTML, SVG `<text>` can be adjusted through custom style sheets. This does not apply for SVG text that is in a separate file, and displayed through, for example, the `img` element.
```

#### Failed Example 2

This image resource referenced by the `input` element in the [Image Button][] contains text for which the particular presentation is not [essential](#0va7u6:essential).

```html
<input type="image" src="/test-assets/0va7u6/button.jpg" alt="Press me" />
```

#### Failed Example 3

This image resource referenced by the `background-image` property of the `div` element contains text for which the particular presentation is not [essential](#0va7u6:essential).

```html
<div style="background-image: url(/test-assets/0va7u6/textimage.jpg); width: 500px; height: 200px;" />
```

#### Failed Example 4

This image resource referenced by the `img` element contains text that provides redundant information, but it still is information, therefore it is not [decorative](#0va7u6:decorative).

```html
<img src="/test-assets/0va7u6/welcome.png" alt="" />
<p>Welcome to our website</p>
```

#### Failed Example 5

> This `img` elements loads an SVG with text as an image resource. Because the SVG is loaded as an image resource, instead of being embedded in HTML the text cannot be selected or customized.

Reviewer 3: ```suggestion
This `img` element loads an SVG with text as an image resource. Because the SVG is loaded as an image resource, instead of being embedded in HTML the text cannot be selected or customized.
```
